server/log_rpc_server_test.go

```diff
@@ -657,35 +698,22 @@ func TestGetLeavesByLeafValueHash(t *testing.T) {
 	}
 }
 
-func TestGetProofByHashBadTreeSize(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
-
-	// Request should fail validation before any storage operations
-	mockStorage := storage.NewMockLogStorage(ctrl)
-	registry := testonly.NewRegistryWithLogProvider(mockStorageProviderFunc(mockStorage))
-	server := NewTrillianLogRPCServer(registry, fakeTimeSource)
-
-	_, err := server.GetInclusionProofByHash(context.Background(), &getInclusionProofByHashRequestBadTreeSize)
+func TestGetProofByHashInvalidRequests(t *testing.T) {
+	for _, test := range hProofReqErrorTests {
+		ctrl := gomock.NewController(t)
 
-	if err == nil {
-		t.Fatalf("get inclusion proof by hash accepted invalid tree size: %v", getInclusionProofByHashRequestBadTreeSize)
-	}
-}
+		// Request should fail validation before any storage operations
+		mockStorage := storage.NewMockLogStorage(ctrl)
+		registry := testonly.NewRegistryWithLogProvider(mockStorageProviderFunc(mockStorage))
+		server := NewTrillianLogRPCServer(registry, fakeTimeSource)
 
-func TestGetProofByHashBadHash(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
+		_, err := server.GetInclusionProofByHash(context.Background(), test.hReq)
```

Comment:
I meant it basically as combining this and the if below, so it wouldn't need any more if/else clauses than we already have.

if _, err := server.GetInclusionProofByHash(context.Background(), test.hReq); err == nil || ... {
	t.Errorf("...")
}

(Just clarifying, please take it as a suggestion.)